src/main/java/com/javax1/cookies/basics/RecipeCollector.java

```diff
@@ -249,6 +300,17 @@ public boolean inspectionIngredient(Stream<Recipe> recipes) {
      */
     @Override
     public Recipe createUltimateRecipe(Stream<Recipe> recipes, Recipe blank) {
-        return null;
+
+        recipes.forEach(r -> r.ingredients().forEach((key, value) -> {
```

Comment:
Nice solution. This is by far the hardest problem in the entire class.
My solution uses `reduce()` but there are at least 3-4 other ways to solve this.

Reply:
my solution to this was very cheaty. I noticed there was a `Recipe.merge` method, so all I did was

```java
recipes.forEach(blank::merge);
return blank;
```

🙈 😄

Reply:
Can you make that 1 line shorter @NPException ? :wink:

Reply:
Ah you're right, could use `.collect`

```java
return recipes.reduce(blank, Recipe::merge);
```